scripts/env.d/devenv_impl

```diff
@@ -17,6 +17,14 @@ nameremove() {
   eval "export $1=$(echo -n ${!1} | awk -v RS=: -v ORS=: -v var="$2" '$0 != var' | sed 's/:*$//')"
 }
 
+# Necessary evil
+if [[ -f "/etc/os-release" ]]; then
+    str=$(grep -o 'ubuntu' /etc/os-release | head -1)
+    if [[ ! -z str ]]; then
```

Comment:
1. Does this line really work? That is to say, should it be `${str}` instead?
2. Is `-n` more intuitive?
 

Besides, (a bit nitpicking), when using variables in the (square) brackets (`[[ ]]` test construct), it is always suggested to use quotes, and `{str}` if possible. For example, `[[ -n "${str}" ]]` may be what you want. See:

```
Surround your variable with " in if [ "${NAME}" = "Kevin" ], because if $NAME isn't declared, bash will throw a syntax error (also see nounset).
```
from https://kvz.io/bash-best-practices.html

Reply:
I concur with @tai271828 .  Use `[ "${NAME}" = "content" ]` instead.

Reply:
Got it, thanks for the feedback.